### PR TITLE
Remove onwards from DotcomRenderingDataModel

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?
This removes the code attempting to move onwards data into agents and return it on the server side to the article request.

This was deemed unsuitable as the data needed was highly queryable (across tags, editions, etc) thus the corpus we would need to store in memory seemed unreasonable.

[The DCR code was removed](https://github.com/guardian/dotcom-rendering/pull/6030) a while back, so this is mostly house keeping.

Ultimately this code won't be used, but the learning from it will. 

This is also taking up a rare 0% test, of which we have just a few.

### CODE REVIEW

**Comment by bryophyta** [comment]
I might well be wrong about this (despite having worked on this originally, my memory is quite fuzzy!), but: I thought that the DCR code hadn't been removed? iirc the code being removed in this PR is used to pass onwards data as part of the Article SSR flow, whereas the endpoint that was removed in [DCR #6030](https://github.com/guardian/dotcom-rendering/pull/6030) was intended to be used by an AJAX endpoint.

It looks like DCR [is still expecting that `onwards` data might be passed](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/layouts/CommentLayout.tsx#L740) as part of the Article data model.

Am I right in thinking that if we're removing the `onwards` field from the `DotcomRenderingDataModel` here then we should also remove it from the `CAPIArticle` type in DCR?

**Comment by jamesgorrie** [comment]
You're completely right @bryophyta - here's the PR that removes that logic from DCR too:
https://github.com/guardian/dotcom-rendering/pull/6894

**Comment by jamesgorrie** [comment]
@bryophyta [the corresponding DCR code has been removed](https://github.com/guardian/dotcom-rendering/pull/6894).

File: common/app/experiments/Experiments.scala [hunk 1]
```diff
@@ -55,18 +55,6 @@ object TableOfContents
...
-object DCROnwardsData
-    extends Experiment(
-      name = "dcr-onwards-data",
-      // DCR will already render the data if we send it down the pipes.
-      // This will allow us to iterate on the feature.
-      // see: https://github.com/guardian/dotcom-rendering/blob/b649a00fc9e5d2ba158f82b7c4b152106579f6a9/dotcom-rendering/src/web/layouts/StandardLayout.tsx#L793-L798
-      description = "Switch to iterate on adding onwards data to be sent to DCR to be server rendered",
-      owners = Seq(Owner.withGithub("@guardian/dotcom-platform")),
-      sellByDate = LocalDate.of(2023, 6, 2),
-      participationGroup = Perc0C,
```
Comment (mxdvl) [inline common/app/experiments/Experiments.scala:67]: cc. @sndrs

**Review by mxdvl — APPROVED** [review]
This seems like a good clean-up!

File: common/app/experiments/Experiments.scala (hunk 1 above)
Comment (jamesgorrie) [reply common/app/experiments/Experiments.scala:67]: cc @bryophyta

**Pull request merged**